File: CSharp/EwtKeepLogin.cs
using System;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;
using System.Timers;

class EwtKeepLogin {
	static void Main(string[] args) {
		var cookies = "";
		if (args.Length > 1) {
			Console.WriteLine("Args > 1!");
			return;
		} else if (args.Length == 0) {
			Console.Write("Please enter your cookies: ");
			cookies = Console.ReadLine();
		} else if (args.Length == 1) {
			if (args[0] == "--help" || args[0] == "-h" || args[0] == "-?") {
				Console.WriteLine("EwtKeepLogin.exe \"you cookies...\"");
				return;
			}
			cookies = args[0];
		}
		var handler = new HttpClientHandler() { UseCookies = false };
		var hc = new HttpClient(handler) { BaseAddress = new Uri("https://web.ewt360.com/customerApi/api/studyprod/lessonCenter/getUserTimeRanking"), Timeout = new TimeSpan(0, 0, 5) };
		hc.DefaultRequestHeaders.Add("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.96 Safari/537.36 Edg/88.0.705.56");
		hc.DefaultRequestHeaders.Add("Cookie", cookies);
		SetInterval(5500, async e => await Send(hc));
		while (true) {
			var input = Console.ReadLine();
			if (input == "quit") {
				Console.WriteLine("Exit!");
				hc.Dispose();
				break;
			} else if (input == "clear") {
				Console.Clear();
			} else {
				Console.WriteLine("Enter \"quit\" then press \"Enter\" to exit.\r\nEnter \"clear\" then press \"Enter\" to clear console.");
			}
		}
	}
	static async Task Send(HttpClient hc) {
		try {
			Console.WriteLine(await hc.GetStringAsync(""));
		} catch (Exception e) {
			Console.Write("Error! Message: ");
			Console.WriteLine(e);
		}
	}
	public static void SetInterval(double interval, Action<ElapsedEventArgs> action) {
		var timer = new Timer(interval);
		timer.Elapsed += delegate (object sender, ElapsedEventArgs e) { action(e); };
		timer.Enabled = true;
	}
}

File: Python/EwtKeepLogin.py
from requests import get
from sys import argv as args
from threading import Thread
from time import sleep, time


def sendMessage(headers):
	while True:
		startTime = time()
		try:
			res = get('https://web.ewt360.com/customerApi/api/studyprod/lessonCenter/getUserTimeRanking', headers=headers, timeout=5)
			print(res.text)
		except Exception as e:
			print('Error! Message: %s' % e)
		endTime = time()
		sleep(5.5 - endTime + startTime)


def keepLogin(cookies):
	headers = {
		"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.96 Safari/537.36 Edg/88.0.705.56",
		"Cookie": cookies
	}
	t = Thread(target=sendMessage, args=(headers,))
	t.setDaemon(True)
	t.start()
	while True:
		cmd = input()
		if cmd == "quit":
			print("Exit!")
			break
		elif cmd == "clear":
			print('\033c', end='')
		else:
			print("Enter \"quit\" to exit.\r\nEnter \"clear\" to clear console.")

if __name__ == '__main__':
	if len(args) >= 1:
		if args[0].split('.')[-1] in ['py', 'pyw']:
			args.pop(0)
	if len(args) > 1:
		print("Args > 1!")
	elif len(args) == 0:
		keepLogin(input("Please enter your cookies: "))
	elif len(args) == 1:
		if args[0] == "--help" or args[0] == "-h" or args[0] == "-?":
			print("EwtKeepLogin.exe \"you cookies...\"")
		else:
			keepLogin(args[0])
